pkg/statsd/handler_cloud.go

```diff
@@ -57,37 +56,50 @@ func (ch *CloudHandler) EstimatedTags() int {
 	return ch.estimatedTags
 }
 
-func (ch *CloudHandler) processMetrics(ctx context.Context, metrics []*gostatsd.Metric) {
+func (ch *CloudHandler) DispatchMetricMap(ctx context.Context, mm *gostatsd.MetricMap) {
 	mmToDispatch := gostatsd.NewMetricMap()
-	var toHandle []*gostatsd.Metric
-	for _, m := range metrics {
-		if ch.updateTagsAndHostname(m, m.Source) {
-			mmToDispatch.Receive(m)
+	mmToHandle := gostatsd.NewMetricMap()
+	mm.Counters.Each(func(metricName string, tagsKey string, c gostatsd.Counter) {
```

Comment:
Should the pointer of the values be passed in here?

Reply:
You mean like:
```
	mm.Counters.Each(func(metricName string, tagsKey string, c *gostatsd.Counter) {
```
...?

If so, no - a MetricMap is all structs, not pointers to structs.